pandas/tseries/period.py

```diff
@@ -211,79 +247,16 @@ def _generate_range(cls, start, end, periods, freq, fields):
         return subarr, freq
 
     @classmethod
-    def _from_arraylike(cls, data, freq, tz):
-        if not isinstance(data, (np.ndarray, PeriodIndex,
-                                 DatetimeIndex, Int64Index)):
-            if lib.isscalar(data) or isinstance(data, Period):
-                raise ValueError('PeriodIndex() must be called with a '
-                                 'collection of some kind, %s was passed'
-                                 % repr(data))
+    def _simple_new(cls, data, name=None, freq=None, **kwargs):
```

Comment:
there *isn't* any current infrastructure, ``Index.__reduce__/__setstate__`` does everything generally. Except for the back-compat stuff (which at some point should just take out). Everythings done as a dict *now* (was a tuple a few versions ago).